explorer-ui/src/components/MatchGrid/AllUsers/AllUsers.jsx

```diff
@@ -0,0 +1,80 @@
+import * as React from "react"
+import "../MatchGrid.css"
+import Geocode from "react-geocode";
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  Flex,
+  HStack,
+  IconButton,
+  Input,
+  SkeletonText,
+  Text,
+} from '@chakra-ui/react'
+Geocode.setApiKey("AIzaSyA4B7q2I3Alla6f8udR0Nr-_3vB8lW5Te0");
+Geocode.setLanguage("en");
+Geocode.setLocationType("ROOFTOP");
+Geocode.enableDebug();
+import MatchCard from "../MatchCard/MatchCard"
+import MapContainer from "../../Mapping/MapContainer";
+import { useState } from "react";
+import { useEffect } from "react";
+import { ChakraProvider, theme } from "@chakra-ui/react";
+export default function AllUsers({profiles}) {
+    const [matchesPlusCoordinates, setMatchesPlusCoordinates] = useState([]);
+    const [userCoordinates, setUserCoordinates] = useState([]);
+    useEffect(() => {
+        getCoordinates(profiles);
+        // getUserCoords(profiles[0])
+    }, [profiles])
+    const getCoordinates = (profiles) => {
+        if (profiles) {
+          for (let i = 0; i < profiles.length; i++) {
+            if(profiles[i].address != null || profiles[i].address!="" ){
```

Comment:
store indexed values as variables to refer to later i.e const profile = profiles[i];

Reply:
used forEach call instead of for loop so no longer a need to keep track of index